cms/server/ContestWebServer.py

```diff
@@ -1061,10 +1063,12 @@ def which_language(user_filename):
                              if it is not a recognized language.
 
             """
-            for source_ext, language in SOURCE_EXT_TO_LANGUAGE_MAP.iteritems():
-                if user_filename.endswith(source_ext):
-                    return language
-            return None
+
+            languageSubmitted = self.get_argument("submission_lang", "")
```

Comment:
Would it be okay to say "HTTP 400: Bad Request (Missing argument submission_lang)"?